src/privileged/messagingSystem/api.js

```diff
@@ -0,0 +1,171 @@
+/* eslint no-console: ["warn", { allow: ["info", "warn", "error"] }] */
+/* global ExtensionAPI */
+
+"use strict";
+
+this.messagingSystem = class extends ExtensionAPI {
+  getAPI(context) {
+    const { Services } = ChromeUtils.import(
+      "resource://gre/modules/Services.jsm",
+      {},
+    );
+
+    const { ExtensionCommon } = ChromeUtils.import(
+      "resource://gre/modules/ExtensionCommon.jsm",
+      {},
+    );
+
+    const { EventManager } = ExtensionCommon;
+
+    const { ExtensionUtils } = ChromeUtils.import(
+      "resource://gre/modules/ExtensionUtils.jsm",
+      {},
+    );
+    const { ExtensionError } = ExtensionUtils;
+
+    const { RemoteSettings } = ChromeUtils.import(
+      "resource://services-settings/remote-settings.js",
+      {},
+    );
+    const { ASRouterTargeting } = ChromeUtils.import(
```

Comment:
If the add-on is still going to run for the control branch but not do much, it would be worth lazily importing all this stuff

Reply:
True, but since web extension experiments are already lazy-loaded and the control branch only invokes this API on first run the impact is manageable. Lazy-loading in experiments has in the past led to a bit harder-to-test code, with some loading errors surfacing late at runtime instead of upfront, for instance when an API has changed between Fx versions.